python/pynq/iop/grove_haptic_motor.py

```diff
@@ -0,0 +1,221 @@
+#
+#   2.  Redistributions in binary form must reproduce the above copyright 
+#       notice, this list of conditions and the following disclaimer in the 
+#       documentation and/or other materials provided with the distribution.
+#
+#   3.  Neither the name of the copyright holder nor the names of its 
+#       contributors may be used to endorse or promote products derived from 
+#       this software without specific prior written permission.
+#
+#   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#   AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, 
+#   THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR 
+#   PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR 
+#   CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, 
+#   EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, 
+#   PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+#   OR BUSINESS INTERRUPTION). HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, 
+#   WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR 
+#   OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF 
+#   ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+__author__      = "Marco Rabozzi, Luca Cerina, Giuseppe Natale"
+__copyright__   = "Copyright 2016, NECST Laboratory, Politecnico di Milano"
+
+import time
+import struct
+from pynq import MMIO
+from pynq.iop import request_iop
+from pynq.iop import iop_const
+from pynq.iop import Pmod_IO
+from pynq.iop import Arduino_IO
+from pynq.iop import PMODA
+from pynq.iop import PMODB
+from pynq.iop import ARDUINO
+from pynq.iop import PMOD_GROVE_G3
+from pynq.iop import PMOD_GROVE_G4
+from pynq.iop import ARDUINO_GROVE_I2C
+
+PMOD_GROVE_HAPTIC_MOTOR_PROGRAM = "pmod_grove_haptic_motor.bin"
+ARDUINO_GROVE_HAPTIC_MOTOR_PROGRAM = "arduino_grove_haptic_motor.bin"
+
+class Grove_Haptic_Motor(object):
+    """This class controls the Grove Haptic Motor based on the DRV2605L.
+    Hardware version v0.9. 
+    
+    
+    Attributes
+    ----------
+    iop : _IOP
+        I/O processor instance used by Grove_Haptic_Motor.
+    mmio : MMIO
+        Memory-mapped I/O instance to read and write instructions and data.
+        
+    """
+    def __init__(self, if_id, gr_pin): 
+        """Return a new instance of an Grove_Haptic_Motor object. 
+                
+        Parameters
+        ----------
+        if_id : int
+            IOP ID (1, 2, 3) corresponding to (PMODA, PMODB, ARDUINO).
+        gr_pin: list
+            A group of pins on stickit connector or arduino shield.
+            
+        """
+        if if_id in [PMODA, PMODB]:
+            if not gr_pin in [PMOD_GROVE_G3, \
+                              PMOD_GROVE_G4]:
+                raise ValueError("Haptic_Motor group number can only be " + \
+                                 "G3 - G4.")
```

Comment:
We choosed to be consisted with your error messages, see for instance [here](https://github.com/Xilinx/PYNQ/blob/master/python/pynq/iop/grove_imu.py#L82). I understand it does not fit in 1 line but this is because of the sensor name. Could you please suggest a proper edit?